### PR TITLE
Add BitTiming class compatibility to Vector interface

### PR DESCRIPTION
**Pull request by zariiii9003**
Implementation based on #625 

BitTiming class takes precedence over the bitrate argument.
I did not remove the other bit timing arguments from __init__() and i did not add any deprecation warnings.

i tested this on a VN1610

### CODE REVIEW

File: can/interfaces/vector/canlib.py [hunk 1]
```diff
@@ -217,20 +224,36 @@ def __init__(
...
+                    self.canFdConf.arbitrationBitRate = ctypes.c_uint(timing.bitrate)
+                    self.canFdConf.sjwAbr = ctypes.c_uint(timing.sjw)
+                    self.canFdConf.tseg1Abr = ctypes.c_uint(timing.tseg1)
+                    self.canFdConf.tseg2Abr = ctypes.c_uint(timing.tseg2)
+                    if data_timing:
+                        self.canFdConf.dataBitRate = ctypes.c_uint(data_timing.bitrate)
+                        self.canFdConf.sjwDbr = ctypes.c_uint(data_timing.sjw)
+                        self.canFdConf.tseg1Dbr = ctypes.c_uint(data_timing.tseg1)
+                        self.canFdConf.tseg2Dbr = ctypes.c_uint(data_timing.tseg2)
+                    else:
```
Comment (hardbyte) [inline can/interfaces/vector/canlib.py]: This `else` clause seems unnecessary - instead why don't we make sure  `data_timing` falls back to the `timing` object

File: can/interfaces/vector/canlib.py [hunk 2]
```diff
@@ -217,20 +224,36 @@ def __init__(
         if permission_mask.value == self.mask:
             if fd:
                 self.canFdConf = xlclass.XLcanFdConf()
-                if bitrate:
-                    self.canFdConf.arbitrationBitRate = ctypes.c_uint(bitrate)
+                if timing:
```
Comment (hardbyte) [inline can/interfaces/vector/canlib.py]: Can we refactor this out of the `init` and get back the `canFdConf` object? That should make the code more readible and make testing the corner cases a bit easier.

**Comment by zariiii9003** [comment]
@hardbyte Thank you for your review! I will take care of it once the discussion in #625 is resolved.

File: can/interfaces/vector/canlib.py (hunk 2 above)
Comment (zariiii9003) [reply can/interfaces/vector/canlib.py]: @hardbyte can you take another look? I implemented the changes you requested.
